### PR TITLE
Automatically create paraview output directory

### PR DESCRIPTION
**Pull request by mschreter**
@nmuch: FYI
... already a long time overdue :-).

### CODE REVIEW

File: include/meltpooldg/utilities/postprocessor.hpp [hunk 1]
```diff
@@ -83,7 +90,7 @@ namespace MeltPoolDG
 
       data_out.build_patches(mapping);
       data_out.write_vtu_with_pvtu_record(pv_data.directory,
-                                          pv_data.filename,
+                                          "/" + pv_data.filename,
```
Comment (peterrum) [inline include/meltpooldg/utilities/postprocessor.hpp:93]: I don't get this change!

As a reference: https://github.com/kronbichler/adaflo/blob/b1b9e319784573f911b8c3306dfa5d2fc5617610/source/flow_base_algorithm.cc#L241-L248

File: include/meltpooldg/utilities/postprocessor.hpp [hunk 2]
```diff
@@ -46,7 +48,12 @@ namespace MeltPoolDG
       , triangulation(triangulation_in)
       , pcout(std::cout, (Utilities::MPI::this_mpi_process(mpi_communicator) == 0))
       , do_simplex(!triangulation.all_reference_cells_are_hyper_cube())
-    {}
+    {
+      // check if the requested paraview directory exists and if not create the directory
+      if (!std::filesystem::is_directory(pv_data.directory) ||
+          !std::filesystem::exists(pv_data.directory))
```
Comment (peterrum) [inline include/meltpooldg/utilities/postprocessor.hpp:54]:
```
      if (!std::filesystem::is_directory(pv_data.directory) &&
          !std::filesystem::exists(pv_data.directory))
```
You might have got a file with the same name!?

**Review by peterrum — APPROVED** [review]
I will not understand what took C++ so long to add `std::filesystem` (C++17).

File: include/meltpooldg/utilities/postprocessor.hpp (hunk 1 above)
Comment (mschreter) [reply include/meltpooldg/utilities/postprocessor.hpp:93]: It won't matter, where the slash is located. I can also add it as a suffix to the `pv_data.directory`.

Comment (peterrum) [reply include/meltpooldg/utilities/postprocessor.hpp:93]: Ah. Yes, let's move it to the folder name. The problem is that the called function does not normalize the names (for that one would need `std::filesystem`).

File: include/meltpooldg/utilities/postprocessor.hpp (hunk 2 above)
Comment (mschreter) [reply include/meltpooldg/utilities/postprocessor.hpp:54]: I removed the second statement.

File: include/meltpooldg/utilities/postprocessor.hpp [hunk 3]
```diff
@@ -46,7 +48,11 @@ namespace MeltPoolDG
       , triangulation(triangulation_in)
       , pcout(std::cout, (Utilities::MPI::this_mpi_process(mpi_communicator) == 0))
       , do_simplex(!triangulation.all_reference_cells_are_hyper_cube())
-    {}
+    {
+      // check if the requested paraview directory exists and if not create the directory
+      if (!std::filesystem::is_directory(pv_data.directory))
+        std::filesystem::create_directory(pv_data.directory);
```
Comment (peterrum) [inline include/meltpooldg/utilities/postprocessor.hpp:54]: Could you create a file with the name `pv_data.directory` and have a look at what happens?

File: include/meltpooldg/utilities/postprocessor.hpp (hunk 2 above)
Comment (mschreter) [reply include/meltpooldg/utilities/postprocessor.hpp]: Mmh...

```
----------------------------------------------------
Exception on processing: 
filesystem error: cannot create directory: File exists [my_directory]
Aborting!
```

Comment (mschreter) [reply include/meltpooldg/utilities/postprocessor.hpp]: ... only if I created a file `my_directory` with `touch my_directory`. If I have a file `my_directory.txt`, it still creates the directory.

File: include/meltpooldg/utilities/postprocessor.hpp (hunk 3 above)
Comment (mschreter) [reply include/meltpooldg/utilities/postprocessor.hpp:54]: Please find the comments above.

File: include/meltpooldg/utilities/postprocessor.hpp [hunk 4]
```diff
@@ -46,7 +48,19 @@ namespace MeltPoolDG
...
-    {}
+    {
+      // check if the requested paraview directory exists and if not create the directory
+      AssertThrow(!std::filesystem::exists(pv_data.directory) ||
+                    std::filesystem::is_directory(pv_data.directory),
+                  ExcMessage("You are trying to create a folder with the name <" +
+                             std::string(pv_data.directory) +
+                             ">. However, a file with the same name already exists! "
+                             "Possible solutions could be to either rename the output "
+                             "folder or to rename/move the existing file."));
```
Comment (peterrum) [inline include/meltpooldg/utilities/postprocessor.hpp:59]: A minor suggestion:
```suggestion
                             "folder in the parameter file or to rename/move the existing file."));
```

**Review by peterrum — APPROVED** [review]
Thanks!

**Pull request merged**